packages/helpers/src/exec-command-wrapper.ts

```diff
@@ -0,0 +1,123 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+
+export class Command {
+  private readonly commandText: string
+  private readonly args: string[]
+  private readonly options: exec.ExecOptions | undefined
+
+  private failOnError = false
+  private throwOnError = false
+
+  private failOnEmptyOutput = false
+  private throwOnEmptyOutput = false
+
+  constructor(
+    commandText: string,
+    args: string[],
+    options: exec.ExecOptions | undefined = undefined
+  ) {
+    this.commandText = commandText
+    this.args = args
+    this.options = options
+  }
+
+  get failOn(): {error: () => Command; empty: () => Command} {
```

Comment:
I think we should change this API to look more conventional. Right now it has calling like this:
```typescript
commandHelper.failOn.error()
```
I think this would be a more traditional approach:
```typescript
commandHelper.onError.fail(/* pass here error with which it would fail or nothing to fail with original error text */)
commandHelper.onError.throw(/* pass here error or text to throw or nothing to throw with original error text */)
commandHelper.onError((error) => /*  leave here a custom error handling */)

commandHelper.onSuccess((output) => /* check here output and act accordingly */)
commandHelper.onOutputEmpty(() => /* actions on empty output */)
commandHelper.onOutputEmpty.fail(/* pass here error with which it would fail or nothing to fail with generic error text */)
commandHelper.onOutputEmpty.throw(/* pass here error or text to throw or nothing to throw with original error text */)
```